src/MessageProcessorFactory.cs

```diff
@@ -25,74 +27,65 @@ namespace Gauge.Dotnet
+            var methodExecutor = new MethodExecutor(sandbox);
+            var handlers = new Dictionary<Message.Types.MessageType, IMessageProcessor>{
+                {Message.Types.MessageType.ExecutionStarting, new ExecutionStartingProcessor(methodExecutor, assemblyLoader, reflectionWrapper)},
+                {Message.Types.MessageType.ExecutionEnding, new ExecutionEndingProcessor(methodExecutor, assemblyLoader, reflectionWrapper)},
+                {Message.Types.MessageType.SpecExecutionStarting, new SpecExecutionStartingProcessor(methodExecutor, sandbox, assemblyLoader, reflectionWrapper) },
+                {Message.Types.MessageType.SpecExecutionEnding,new SpecExecutionEndingProcessor(methodExecutor, sandbox, assemblyLoader, reflectionWrapper)},
+                {Message.Types.MessageType.ScenarioExecutionStarting,new ScenarioExecutionStartingProcessor(methodExecutor, sandbox, assemblyLoader, reflectionWrapper)},
+                {Message.Types.MessageType.ScenarioExecutionEnding,new ScenarioExecutionEndingProcessor(methodExecutor, sandbox, assemblyLoader, reflectionWrapper)},
+                {Message.Types.MessageType.StepExecutionStarting, new StepExecutionStartingProcessor(methodExecutor, assemblyLoader, reflectionWrapper)},
+                {Message.Types.MessageType.StepExecutionEnding, new StepExecutionEndingProcessor(methodExecutor, assemblyLoader, reflectionWrapper)},
+                {Message.Types.MessageType.ExecuteStep, new ExecuteStepProcessor(_stepRegistry,methodExecutor, tableFormatter)},
                 {Message.Types.MessageType.KillProcessRequest, new KillProcessProcessor()},
-                {Message.Types.MessageType.StepNamesRequest, new StepNamesProcessor(_stepScanner)},
-                {Message.Types.MessageType.StepValidateRequest, new StepValidationProcessor(stepRegistry)},
-                {Message.Types.MessageType.ScenarioDataStoreInit, new ScenarioDataStoreInitProcessor(_assemblyLoader)},
-                {Message.Types.MessageType.SpecDataStoreInit, new SpecDataStoreInitProcessor(_assemblyLoader)},
-                {Message.Types.MessageType.SuiteDataStoreInit, new SuiteDataStoreInitProcessor(_assemblyLoader)},
-                {Message.Types.MessageType.StepNameRequest, new StepNameProcessor(stepRegistry)},
-                {Message.Types.MessageType.RefactorRequest, new RefactorProcessor(stepRegistry, _sandbox)}
+                {Message.Types.MessageType.ScenarioDataStoreInit, new ScenarioDataStoreInitProcessor(assemblyLoader)},
+                {Message.Types.MessageType.SpecDataStoreInit, new SpecDataStoreInitProcessor(assemblyLoader)},
+                {Message.Types.MessageType.SuiteDataStoreInit, new SuiteDataStoreInitProcessor(assemblyLoader)},
             };
-            return messageHandlers;
+            handlers.ToList().ForEach(x => _messageProcessorsDictionary.Add(x.Key, x.Value));
```

Comment:
you do not need to do a `ToList` on a dictionary.

Reply:
done